data/transactions/logic/eval.go

```diff
@@ -1872,6 +1890,15 @@ func (cx *evalContext) txnFieldToStack(txn *transactions.Transaction, field TxnF
 		sv.Uint = txn.Fee.Raw
 	case FirstValid:
 		sv.Uint = uint64(txn.FirstValid)
+	case FirstValidTime:
+		// Check if the first valid time has already been set
+		if FirstValidTime == 0 {
```

Comment:
Set the tx field here, if it has not been initialized already (i.e. `FirstValidTime` was not called already` - this field should not be set by the user anyway)